docs/config.md

```diff
@@ -13,57 +15,100 @@ body of the hosting page:
+JavaScript:
+
+```html
+window.hypothesisConfig = function () {
+  return {
+    "openSidebar": true
+  };
+};
+```
+
+Config settings
+---------------
 
-## Client behavior
+### Client behavior
 
-These keys configure the behavior and initial state of the client when it
+These settings configure the behavior and initial state of the client when it
 loads.
 
-### `openLoginForm`
+#### `openLoginForm`
 
 _Boolean_. Controls whether the login panel is automatically opened on startup,
 as if the user had clicked "Log in" themselves. (Default: _false_.)
 
-### `openSidebar`
+#### `openSidebar`
 
 _Boolean_. Controls whether the sidebar opens automatically on startup.
 (Default: _false_.)
 
-### `showHighlights`
+#### `showHighlights`
 
 _Boolean_. Controls whether the in-document highlights are shown by default.
 (Default: _true_.)
 
-## Annotation services
+### Annotation services
 
-These keys configure which annotation services the client connects to and where
+These settings configure which annotation services the client connects to and where
 it loads assets from. By default, the client will connect to the public
 Hypothesis service at [hypothes.is](https://hypothes.is).
 
-### `assetRoot`
+#### `assetRoot`
 
 _String_. The URL from which client assets are loaded.
 
-### `sidebarAppUrl`
+#### `sidebarAppUrl`
 
 _String_. The URL for the sidebar application which displays annotations
 (Default: _https://hypothes.is/app.html_).
 
-### `services`
+#### `services`
 
 _Array_. A list of additional annotation services which the client should
 retrieve annotations from, optionally including information about the identity
 of the user on that service. This list is in addition to the public
 [Hypothesis](https://hypothes.is/) service.
 
-Each service description is an object with the keys:
+**N.B.** Currently only one additional annotation service is supported - only
+the first item in this `services` array is used, and any further items in the
+array are ignored.
+
+Each item in the array should be an object describing an annotation service,
+with the following keys:
 
  * `authority` _String_. The domain name which the annotation service is associated with.
+
  * `grantToken` _String|null_. An OAuth grant token which the client can exchange for an access token in order to make authenticated requests to the service. If _null_, the user will only be able to read rather than create or modify annotations. (Default: _null_)
+
  * `icon` _String|null_. The URL to an image for the annotation service. This image will appear to the left of the name of the currently selected group. The image should be suitable for display at 16x16px and the recommended format is SVG.
+
+ * `onLogin` _function_. A JavaScript function that the Hypothesis client will
+   call in order to login (for example, when the user clicks a login button in
+   the Hypothesis client's sidebar).
+
+   This setting can only be [set using JavaScript](#configuring-the-client-using-javascript).
```

Comment:
+1 on adding this. However, I don't know that "this can only be set using javascript" is all that obvious. I would suggest saying, "only available using `window.hypothesisConfig`" or something more obvious